pkg/ccl/crosscluster/logical/logical_replication_job.go

```diff
@@ -311,15 +312,45 @@ func (p *logicalReplicationPlanner) generatePlanWithFrontier(
 		return nil, nil, nil, err
 	}
 
-	dstToSrcDescMap := make(map[int32]descpb.TableDescriptor)
-	tableIDs := make([]int32, len(p.payload.ReplicationPairs))
-	for i, pair := range p.payload.ReplicationPairs {
-		dstToSrcDescMap[pair.DstDescriptorID] = plan.DescriptorMap[pair.SrcDescriptorID]
-		tableIDs[i] = pair.DstDescriptorID
+	tableIDToName := make(map[int32]fullyQualifiedTableName)
+	tablesMd := make(map[int32]execinfrapb.TableReplicationMetadata)
+	if err := sql.DescsTxn(ctx, execCfg, func(ctx context.Context, txn isql.Txn, descriptors *descs.Collection) error {
+		for _, pair := range p.payload.ReplicationPairs {
+			srcTableDesc := plan.DescriptorMap[pair.SrcDescriptorID]
+
+			// Look up fully qualified destination table name
+			dstTableDesc, err := descriptors.ByID(txn.KV()).WithoutNonPublic().Get().Table(ctx, descpb.ID(pair.DstDescriptorID))
+			if err != nil {
+				return errors.Wrapf(err, "failed to look up table descriptor %d", pair.DstDescriptorID)
+			}
+			dbDesc, err := descriptors.ByID(txn.KV()).WithoutNonPublic().Get().Database(ctx, dstTableDesc.GetParentID())
+			if err != nil {
+				return errors.Wrapf(err, "failed to look up database descriptor for table %d", pair.DstDescriptorID)
+			}
+			scDesc, err := descriptors.ByID(txn.KV()).WithoutNonPublic().Get().Schema(ctx, dstTableDesc.GetParentSchemaID())
+			if err != nil {
+				return errors.Wrapf(err, "failed to look up schema descriptor for table %d", pair.DstDescriptorID)
+			}
+
+			tablesMd[pair.DstDescriptorID] = execinfrapb.TableReplicationMetadata{
+				SourceDescriptor:              srcTableDesc,
+				DestinationParentDatabaseName: dbDesc.GetName(),
+				DestinationParentSchemaName:   scDesc.GetName(),
+				DestinationTableName:          dstTableDesc.GetName(),
+			}
+			tableIDToName[pair.DstDescriptorID] = fullyQualifiedTableName{
+				database: dbDesc.GetName(),
+				schema:   scDesc.GetName(),
+				table:    dstTableDesc.GetName(),
+			}
+		}
+		return nil
+	}); err != nil {
+		return nil, nil, nil, err
 	}
 
-	dlqClient := InitDeadLetterQueueClient(p.jobExecCtx.ExecCfg().InternalDB.Executor())
-	if err := dlqClient.Create(ctx, tableIDs); err != nil {
+	dlqClient := InitDeadLetterQueueClient(p.jobExecCtx.ExecCfg().InternalDB.Executor(), tableIDToName)
+	if err := dlqClient.Create(ctx); err != nil {
```

Comment:
It isn't a _problem_ but this will end up trying to recreate the table on every single call of this function by the periodic plan checker.  A quick win here might be to add a flag to this function that we could pass in the replanner to avoid this extra work.

Reply:
I think the logic would have to be a bit more complicated, given that I imagine schema changes (creating or renaming a table in the target) would require a new dlq table. Perhaps we can chat about this in -eng discussion today.

Reply:
@msbutler Yes, but note that the plan made by the replan _checker_  never gets used. Rather it is just used to return an error. So my proposal is that the call made by the replan _checker_ would pass the "don't do extra work" flag and everything else would pass the "do all the work" flag.

Reply:
ooooh, yeah you're right.

Reply:
I'll need to look into the replanning logic a bit more, will address this and the comment above in another pr!